Mage/src/main/java/mage/abilities/effects/common/SurveilXEffect.java

```diff
@@ -0,0 +1,52 @@
+package mage.abilities.effects.common;
+
+import mage.abilities.Ability;
+import mage.abilities.dynamicvalue.DynamicValue;
+import mage.abilities.dynamicvalue.common.StaticValue;
+import mage.abilities.effects.OneShotEffect;
+import mage.constants.Outcome;
+import mage.game.Game;
+import mage.players.Player;
+
+public class SurveilXEffect extends OneShotEffect {
```

Comment:
it would be better to build the x support into surveil effect but I think that would be better to do in a separate refactor

Reply:
I never know when to modify other peoples code. After I'm done with cards I'll have a look at the refector

Reply:
alright then in that case you should get rid of this effect and just have a custom effect in the main card file

Reply:
I don't think we can write the X effect into surveil because of the getText function. All the other Scry X, Surveil X have the function written as a one off, it might be nice to just leave it unless you really don't want it I can move to be a custom effect

Reply:
If you're gonna create a separate effect then you should refactor other instances that would use that effect, otherwise create a custom effect within the file. We can definitely modify the text generation of `SurveilEffect` to handle dynamic values, there are already plenty of examples of effects that do this. I think we should leave it for a separate refactor at this point as there are only three cards that currently surveil X.

Reply:
(A unit test would be useful)

Reply:
I moved it back into a custom effect. Did you still want a unit test?

Reply:
The advantage of a unit test is to ensure it works after a future refactor. So while not strictly required, yes, I would like one.

Reply:
I wasn't going to refactor SurveilEffect

Reply:
yes I understand but someone will eventually